File: Python/Problem2.py
class Solution:
    def minPatches(self, nums: List[int], n: int) -> int:
        patches = 0
        missingNumber = 1
        i = 0
        while missingNumber <= n:
            if i<len(nums) and nums[i] <= missingNumber:
                missingNumber += nums[i]
                i += 1
            else:
                patches += 1
                missingNumber += missingNumber
        return patches

File: Typescript/Problem2.ts
function minPatches(nums: number[], n: number): number {
    let patches = 0;
    let missingNumber = 1;
    let i = 0;
    
    while (missingNumber <= n) {
        if (i < nums.length && nums[i] <= missingNumber) {
            missingNumber += nums[i];
            i++;
        } else {
            patches++;
            missingNumber += missingNumber;
        }
    }
    return patches;
};